src/main/java/io/stargate/sgv2/jsonapi/api/model/command/builders/SortClauseBuilder.java

```diff
@@ -64,178 +57,59 @@ public SortClause build(JsonNode node) {
 
   protected abstract SortClause buildAndValidate(ObjectNode sortNode);
 
+  protected abstract String validateSortClausePath(String path);
+
   protected SortClause defaultBuildAndValidate(ObjectNode sortNode) {
     // safe to iterate, we know it's an Object
     Iterator<Map.Entry<String, JsonNode>> fieldIter = sortNode.fields();
     int totalFields = sortNode.size();
     List<SortExpression> sortExpressions = new ArrayList<>(sortNode.size());
 
-    // $lexical is only allowed alone: handle first
-    JsonNode lexicalValue = sortNode.get(DocumentConstants.Fields.LEXICAL_CONTENT_FIELD);
-    if (lexicalValue != null) {
-      if (sortNode.size() > 1) {
-        throw ErrorCodeV1.INVALID_SORT_CLAUSE.toApiException(
-            "if sorting by '%s' no other sort expressions allowed",
-            DocumentConstants.Fields.LEXICAL_CONTENT_FIELD);
-      }
-      if (!lexicalValue.isTextual()) {
-        throw ErrorCodeV1.INVALID_SORT_CLAUSE.toApiException(
-            "if sorting by '%s' value must be String, not %s",
-            DocumentConstants.Fields.LEXICAL_CONTENT_FIELD,
-            JsonUtil.nodeTypeAsString(lexicalValue));
-      }
-      // We cannot yet determine if lexical sort supported by the collection, just
-      // construct clause
-      return new SortClause(
-          Collections.singletonList(SortExpression.bm25Search(lexicalValue.textValue())));
-    }
-
     while (fieldIter.hasNext()) {
       Map.Entry<String, JsonNode> inner = fieldIter.next();
       final String path = inner.getKey().trim();
       // Validation will check against invalid paths, as well as decode "amp-escaping"
-      final String validatedPath = validateSortClausePath(path);
-
-      float[] vectorFloats = null;
```

Comment:
This whole code extracted into per-expression method, separate for Table/Collection builders.